ui/k-toolbar/main.kytos

```diff
@@ -41,6 +55,171 @@
+                                          "description": self.description,
+                                          "start": self.start,
+                                          "end": self.end,
+                                          "items": self.merged_list,         
+                    }),
+                    url:this.$kytos_server_api + "kytos/maintenance",
+                });
+
+                request.done(function() {
+                    let notification = {
+                        icon: 'gear',
+                        title: 'Maintenance Window Created',
+                        description: 'Maintenance Window with id ' + self.id + ' was created.'
+                    }
+                    self.$kytos.$emit("setNotification", notification);
+                }),
+                request.fail(function(jqXHR, status, error) {
+
+                    error_message = JSON.parse(jqXHR.responseText)
+                    if (error_message.hasOwnProperty('response')) {
+                        error_message = error_message.response
+                    } else if (error_message.hasOwnProperty('description')) {
+                        error_message = error_message.description
+                    }
+
+                    let notification = {
+                        icon: 'gear',
+                        title: 'Maintenance Window not created',
+                        description: 'Maintenance Window with id ' + self.id + ' was not created. Error: ' + error_message
+                    }
+                   self.$kytos.$emit("setNotification", notification);
+               }),
+             self.merged_list = [];
+            },
+            /*
+                Gets the Switch and Interface information using the GET api call from topology. 
+            */
+            loadNames: function() {
+                let _dpid_names = [];
+                let _interface_names = [];
+                var _this = this;
+
+                var request = $.ajax({
+                    url: this.$kytos_server_api + "kytos/topology/v3/switches",
+                    type: "GET",
+                    data: JSON.stringify(),
+                    dataType: "json",
+                    contentType: "application/json; charset=utf-8"
+               });
+
+               request.done(function(data) {
+                   let switches = data.switches;
+                   $.each(switches, function(i, sw) {
+                       if(sw.metadata.node_name){
+                           _dpid_names.push({"value": sw.id, "description": sw.metadata.node_name})
+                       }
+                       else{
+                           _dpid_names.push({"value": sw.id, "description": sw.id})
+                       }
+                       if(sw.interfaces){
+                           $.each(sw.interfaces, function(j, interface){
+                           if(Object.keys(interface.metadata).length == 0){
```

Comment:
@rohanCodes001 this conditional can lead to issues, metadata length can be different than zero and also not have `port_name` defined.  Could you make this conditional safer, something analogous that you've done when checking for switch metadata.node_name?